lib/LaTeXML/resources/XSLT/LaTeXML-common.xsl

```diff
@@ -81,7 +81,7 @@
     <xsl:if test="//ltx:date[@role='creation' or @role='conversion'][1]">
       <xsl:comment>
         <xsl:text>Document created on </xsl:text>
-        <xsl:value-of select='//ltx:date/node()'/>
+        <xsl:value-of select='translate(//ltx:date/node(),"--","—")'/>
         <xsl:text>.</xsl:text>
```

Comment:
the trailing dot ensures we never end a comment with a trailing dash, which is neat, only had to address the double-dash case.